java/flight/src/main/java/org/apache/arrow/flight/grpc/StatusUtils.java

```diff
@@ -0,0 +1,192 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.arrow.flight.grpc;
+
+import java.util.Iterator;
+import java.util.Objects;
+import java.util.function.Function;
+
+import org.apache.arrow.flight.CallStatus;
+import org.apache.arrow.flight.FlightRuntimeException;
+import org.apache.arrow.flight.FlightStatusCode;
+
+import io.grpc.Status;
+import io.grpc.Status.Code;
+import io.grpc.StatusException;
+import io.grpc.StatusRuntimeException;
+
+/**
+ * Utilities to adapt gRPC and Flight status objects.
+ *
+ * <p>NOT A PUBLIC CLASS, interface is not guaranteed to remain stable.
+ */
+public class StatusUtils {
+
+  private StatusUtils() {
+    throw new AssertionError("Do not instantiate this class.");
+  }
+
+  /**
+   * Convert from a Flight status code to a gRPC status code.
+   */
+  public static Status.Code toGrpcStatusCode(FlightStatusCode code) {
+    switch (code) {
+      case OK:
+        return Code.OK;
+      case UNKNOWN:
+        return Code.UNKNOWN;
+      case INTERNAL:
+        return Code.INTERNAL;
+      case INVALID_ARGUMENT:
+        return Code.INVALID_ARGUMENT;
+      case TIMED_OUT:
+        return Code.DEADLINE_EXCEEDED;
+      case CANCELLED:
+        return Code.CANCELLED;
+      case UNAUTHENTICATED:
+        return Code.UNAUTHENTICATED;
+      case UNAUTHORIZED:
+        return Code.PERMISSION_DENIED;
+      case UNIMPLEMENTED:
+        return Code.UNIMPLEMENTED;
+      case UNAVAILABLE:
+        return Code.UNAVAILABLE;
+      default:
+        return Code.UNKNOWN;
+    }
+  }
+
+  /**
+   * Convert from a gRPC status code to a Flight status code.
+   */
+  public static FlightStatusCode fromGrpcStatusCode(Status.Code code) {
+    switch (code) {
+      case OK:
+        return FlightStatusCode.OK;
+      case CANCELLED:
+        return FlightStatusCode.CANCELLED;
+      case UNKNOWN:
+        return FlightStatusCode.UNKNOWN;
+      case INVALID_ARGUMENT:
+        return FlightStatusCode.INVALID_ARGUMENT;
+      case DEADLINE_EXCEEDED:
+        return FlightStatusCode.TIMED_OUT;
+      case NOT_FOUND:
+        return FlightStatusCode.INVALID_ARGUMENT;
+      case ALREADY_EXISTS:
+        return FlightStatusCode.INVALID_ARGUMENT;
```

Comment:
per comment above these still seem like INVALID_ARGUMENT might be too broad a catch-all.

Reply:
I added these two codes to the Flight codes, to mirror what was done in C++/Python.